modules/backend/widgets/Lists.php

```diff
@@ -1698,6 +1710,84 @@ public function onToggleTreeNode()
         return $this->onRefresh();
     }
 
+    //
+    // Internal
+    //
+
+    /**
+     * Looks at the model for defined options.
+     *
+     * @param $field
+     * @param $fieldOptions
+     * @return mixed
+     */
+    protected function getOptionsFromModel($field, $fieldOptions, $fieldName)
+    {
+        /*
+         * Advanced usage, supplied options are callable
+         */
+        if (is_array($fieldOptions) && is_callable($fieldOptions)) {
```

Comment:
@SaifurRahmanMohsin Could you just check if `$fieldOptions` is "callable" here?
```suggestion
        if (is_callable($fieldOptions)) {
```

Reply:
Yep, this works if you define the options field in the list fields config as a reference to a method in the model that provides the array of options.

Reply:
@bennothommo doesn't that have the potential to backfire (or at the very least introduce yet another way to load options), say if someone specified `options: global_function_here`?

Reply:
@LukeTowers The same could happen with an array too. Perhaps this whole block should go.

Reply:
@bennothommo $fieldOptions is callable if you specify it in an array format as mentioned [here.](/octobercms/docs/pull/373/files)

Reply:
@SaifurRahmanMohsin Thanks for that, you are correct. You can leave your code as is.

Reply:
Ok @bennothommo. Thanks anyway for mentioning this coz I realized that there is a 5th way of specifying the options that hasn't been documented.